bundle/Command/ScheduledVisibilityUpdateCommand.php

```diff
@@ -0,0 +1,248 @@
+        );
+
+        if (!$this->style->askQuestion($question)) {
+            $this->style->success('Aborted');
+
+            return Command::SUCCESS;
+        }
+
+        if (!$this->configurationService->isEnabled()) {
+            $this->style->warning('Scheduled visibility mechanism is disabled.');
+
+            return Command::FAILURE;
+        }
+
+        $allContentTypes = $this->configurationService->isAllContentTypes();
+        $allowedContentTypes = $this->configurationService->getAllowedContentTypes();
+
+        if (!$allContentTypes && count($allowedContentTypes) === 0) {
+            $this->style->warning('No content types configured for scheduled visibility mechanism.');
+
+            return Command::FAILURE;
+        }
+
+        $pager = $this->getPager($allContentTypes, $allowedContentTypes);
+
+        if ($pager->getNbResults() === 0) {
+            $this->style->info('No content found');
+
+            return Command::FAILURE;
+        }
+
+        $limit = $input->getOption('limit');
+        $offset = 0;
+
+        $this->style->createProgressBar($pager->getNbResults());
+        $this->style->progressStart();
+        $this->style->newLine();
+
+        $results = $pager->getAdapter()->getSlice($offset, $limit);
+        while (count($results) > 0) {
+            $this->processResults($results);
+            $offset += $limit;
+            $results = $pager->getAdapter()->getSlice($offset, $limit);
+        }
+
+        $this->style->progressFinish();
+
+        $this->style->info('Done.');
+
+        return Command::SUCCESS;
+    }
+
+    private function processResults(array $results): void
+    {
+        foreach ($results as $result) {
+            try {
+                $languageId = $result['initial_language_id'];
+                $language = $this->loadLanguage($languageId);
+            } catch (NotFoundException $exception) {
+                $this->logger->error(
+                    sprintf(
+                        'Language with id #%d does not exist: %s',
+                        $languageId,
+                        $exception->getMessage(),
+                    ),
+                );
+
+                continue;
+            }
+
+            try {
+                $contentId = $result['id'];
+
+                /** @var \Ibexa\Contracts\Core\Repository\Values\Content\Content $content */
+                $content = $this->repository->sudo(
+                    fn () => $this->repository->getContentService()->loadContent(
+                        $contentId,
+                        [$language->getLanguageCode()],
+                    ),
+                );
+            } catch (NotFoundException $exception) {
+                $this->logger->error(
+                    sprintf(
+                        'Content with id #%d does not exist: %s',
+                        $contentId,
+                        $exception->getMessage(),
+                    ),
+                );
+
+                continue;
```

Comment:
Progress bar still needs to be incremented as the item is part of the total count.

Reply:
fixed in 0b364a8056da4759948537d0b8f9d90e9f81c4de